### ISSUE TITLE
PCC

### ISSUE THREAD

**Issue by yegor256** [yegor256/jpeek#67]
Let's implement PCC cohesion metric:

 C. Bonja and E. Kidanmariam, “Metrics for class cohesion and
similarity between methods,” in Proceedings of the 44th annual
Southeast regional conference. ACM, 2006, pp. 91–95.

https://dl.acm.org/citation.cfm?id=1185469 (didn't find PDF available for free).

**Comment by 0crat** [comment]
Job [#67](https://github.com/yegor256/jpeek/issues/67) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://datum.zerocracy.com/pages/policy.html#29): +15 points just awarded to [@yegor256](http://www.0crat.com/u/yegor256), total is [+7225](http://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#67](https://github.com/yegor256/jpeek/issues/67) assigned to @flslkxtc[/z](http://www.0crat.com/u/flslkxtc). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by 0crat** [comment]
The job [#67](https://github.com/yegor256/jpeek/issues/67) assigned to @driver733[/z](https://www.0crat.com/u/driver733). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by driver733** [comment]
@0crat waiting

**Comment by driver733** [comment]
Will get back to the task after implementation of other algorithms

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/jpeek/issues/67#issuecomment-361813744))

@driver733 The impediment for [#67](https://github.com/yegor256/jpeek/issues/67) was registered successfully by @driver733[/z](https://www.0crat.com/u/driver733)

**Comment by 0crat** [comment]
The job [#67](https://github.com/yegor256/jpeek/issues/67) assigned to @gizmomogwai[/z](https://www.0crat.com/u/gizmomogwai). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by 0crat** [comment]
@gizmomogwai[/z](https://www.0crat.com/u/gizmomogwai) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The job [#67](https://github.com/yegor256/jpeek/issues/67) assigned to @llorllale[/z](https://www.0crat.com/u/llorllale), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/401523081). The budget is 30 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§8](http://datum.zerocracy.com/pages/policy.html#8) and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by llorllale** [comment]
I've just requested the full text of the paper and put "jpeek (https://github.com/yegor256/jpeek)" as my "institution or company" as per their form.

**Comment by llorllale** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/jpeek/issues/67#issuecomment-364470861))

@llorllale The impediment for [#67](https://github.com/yegor256/jpeek/issues/67) was registered successfully by @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by llorllale** [comment]
@yegor256 isn't PCC = "Pesimistic Class Cohesion", defined in the `aman04.pdf` paper on page 5?

**Comment by llorllale** [comment]
@yegor256 please confirm the above

**Comment by yegor256** [comment]
@llorllale yes, it is.

**Comment by llorllale** [comment]
@yegor256 great, I'll work with that then

**Comment by llorllale** [comment]
@0crat waiting on reviewer, see [this](https://github.com/yegor256/jpeek/pull/168#issuecomment-365260683)

**Comment by 0crat** [comment]
> @0crat waiting on reviewer, see [this](https://github.com/yegor256/jpeek/pull/168#issuecomment-36... ([here](https://github.com/yegor256/jpeek/issues/67#issuecomment-365263165))

@llorllale Job [#67](https://github.com/yegor256/jpeek/issues/67) is already on hold

**Comment by llorllale** [comment]
@yegor256 this was fixed in #173 (with puzzle #205 opened to finish implementing tests)

Can we close this?

**Comment by yegor256** [comment]
@llorllale thanks

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@driver733 According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your [message](https://github.com/yegor256/jpeek/issues/67#issuecomment-361813795) by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@llorllale According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your [message](https://github.com/yegor256/jpeek/issues/67#issuecomment-364470831) by indicating an addressee in the beginning.

**Comment by llorllale** [comment]
@ypshenychka fixed message

**Comment by ypshenychka** [comment]
@llorllale Thank you

**Comment by llorllale** [comment]
@ypshenychka 
@driver733 is no longer working on this task, therefore has no incentive to come back and fix his messages

**Comment by ypshenychka** [comment]
@llorllale You as a performer of ticket is interested that everything is in accrodance to QA rules. Pay attention to this in future. I will give some time @driver733 to fix this.

**Comment by ypshenychka** [comment]
@driver733 ping. Please consider the above [remark](https://github.com/yegor256/jpeek/issues/67#issuecomment-366818155)

**Comment by llorllale** [comment]
@ypshenychka I am interested, but I cannot do anything about messages from others.

**Comment by ypshenychka** [comment]
@llorllale You could reach them earlier ("in hot pursuit") and ask to correct the message. Probably it would work :) But that's definitely just an advise for a future. If you fix all issues prior to QA review, it can be a good quality ticket with +5 bonus points.

**Comment by llorllale** [comment]
@ypshenychka good advise, but that still doesn't guarantee anything. For example, I've worked on issues before on which people *who aren't even zerocrats* have left messages long ago. 

I cannot be held responsible for the state the issue was left in before I got it.

**Comment by ypshenychka** [comment]
@llorllale I understand. But @driver733 is still a member of a project and he can make correction. Usually I give few days to react. If no - well, we should leave as is...

**Comment by driver733** [comment]
@ypshenychka Fixed.

**Comment by ypshenychka** [comment]
@driver733 Thanks a lot!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+420](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +30 points just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale), total is [+745](https://www.0crat.com/u/llorllale)

**Comment by 0pdd** [comment]
@yegor256 the only puzzle [#205](https://github.com/yegor256/jpeek/issues/205) is solved here.
